lib/vehicle.rb

```diff
@@ -5,14 +5,18 @@ class Vehicle
               :year,
               :make,
               :model,
-              :engine
+              :engine,
+              :registration_date
+              
 
   def initialize(vehicle_details)
     @vin = vehicle_details[:vin]
     @year = vehicle_details[:year]
     @make = vehicle_details[:make]
     @model = vehicle_details[:model]
     @engine = vehicle_details[:engine]
+    @registration_date = vehicle_details[:registration_date]
+    @plate_type = vehicle_details[:plate_type]
```

Comment:
I wouldn't put this as an instance variable in your `initialize` method because it is less dynamic. I would just use the `plate_type` method that you have below.